apps/src/code-studio/initApp/project.js

```diff
@@ -531,32 +532,23 @@ var projects = module.exports = {
    * @param {boolean} forceNewVersion If true, explicitly create a new version.
    * @param {boolean} preparingRemix Indicates whether this save is part of a remix.
    */
-  save(sourceAndHtml, callback, forceNewVersion, preparingRemix) {
+  save(...args) {
+    let [sourceAndHtml, callback, forceNewVersion, preparingRemix] = args;
     // Can't save a project if we're not the owner.
     if (current && current.isOwner === false) {
       return;
     }
 
-    $('.project_updated_at').text('Saving...');  // TODO (Josh) i18n
-
-    if (typeof arguments[0] === 'function' || !sourceAndHtml) {
-      // If no source is provided, shift the arguments and ask for the source
-      // ourselves.
-      var args = Array.prototype.slice.apply(arguments);
-      callback = args[0];
-      forceNewVersion = args[1];
-      preparingRemix = args[2];
-
-      let completeAsyncSave = function () {
-        this.sourceHandler.getAnimationList(animations => {
-          this.sourceHandler.getLevelSource().then(response => {
-            const source = response;
-            const html = this.sourceHandler.getLevelHtml();
-            const makerAPIsEnabled = this.sourceHandler.getMakerAPIsEnabled();
-            this.save({source, html, animations, makerAPIsEnabled}, callback, forceNewVersion);
-          });
-        });
-      }.bind(this);
+    $('.project_updated_at').text(msg.saving());
+
+    if (typeof args[0] === 'function' || !sourceAndHtml) {
+      // If no save data is provided, shift the arguments and ask for the
+      // latest save data ourselves.
+      [callback, forceNewVersion, preparingRemix] = args;
```

Comment:
I like how this ended up working out :) (The arg shift is much more understandable this way I think).